cpp/daal/include/algorithms/covariance/covariance_types.h

```diff
@@ -126,8 +126,9 @@ class DAAL_EXPORT InputIface : public daal::algorithms::Input
 {
 public:
     InputIface(size_t nElements) : daal::algorithms::Input(nElements) {}
-    InputIface(const InputIface & other) : daal::algorithms::Input(other) {}
-    virtual size_t getNumberOfFeatures() const = 0;
+    InputIface(const InputIface & other)             = default;
```

Comment:
This sort of thing (adding definitions in headers that get compiled for different instruction sets) was causing 'illegal intruction' errors in the CI:
https://github.com/uxlfoundation/oneDAL/pull/3012

Reply:
It looks like this method (L130) wasn't there before:
```cpp
InputIface & operator=(const InputIface & other) = default;
```

Reply:
Yeah, but copy-constructor was already defined in header, so I think defining copy-assignment operator here wouldn't change anything

Reply:
Perhaps @Vika-F could provide some perspective here.